### PR TITLE
Extracted CleanAppBar & ScrollableAppBar from CustomAppBar

### PR DESCRIPTION
**Pull request by chinkysight**

### CODE REVIEW

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      if (widget.pageScrollController.position.pixels <=
+          scrollController.position.maxScrollExtent) {
+        scrollController.jumpTo(widget.pageScrollController.position.pixels);
+      }
+
+      if (widget.pageScrollController.position.pixels >
+          widget.pageScrollController.position.minScrollExtent) {
+        if (widget.elevation == null) {
+          if (currentElevation == widget.minElevation) {
+            setState(() {
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:59]: We can use shorthand syntax here `=>`

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+              currentElevation = widget.maxElevation;
+            });
+          }
+        }
+      } else if (widget.pageScrollController.position.pixels ==
+          widget.pageScrollController.position.minScrollExtent) {
+        if (widget.elevation == null) {
+          setState(() {
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:67]: We can use shorthand syntax here `=>`

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      data: Theme.of(context).copyWith(
+          elevatedButtonTheme: ElevatedButtonThemeData(
+              style: ButtonStyle(
+        overlayColor: MaterialStateProperty.resolveWith(
+          (states) => kAlmostTransparent,
+        ),
+        elevation: MaterialStateProperty.all<double>(0.0),
+        shape: MaterialStateProperty.all<OutlinedBorder>(const CircleBorder()),
+        backgroundColor: MaterialStateProperty.all<Color>(Colors.white),
+      ))),
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:93]: Can we add a few `,` here for better formatting

File: lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          leading: leading,
+          actions: actions,
+          title: Text(title),
+          titleTextStyle: titleTextStyle ??
+              Theme.of(context)
+                  .textTheme
+                  .headline6
+                  ?.copyWith(color: kPrimaryColor),
+          backwardsCompatibility: false,
+        ));
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart:50]: Add a `,` here for better formatting

File: lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  @override
+  Size get preferredSize => const Size(double.infinity, 56.0);
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart:54]: Can we replace 56.0 with `kToolbarHeight` or is it not entirely the same?

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      this.minElevation = 0.0,
+      this.maxElevation = 2.0,
+      required this.pageScrollController})
+      : super(key: key);
+
+  @override
+  ScrollableAppBarState createState() => ScrollableAppBarState();
+
+  @override
+  Size get preferredSize => const Size(double.infinity, 56.0);
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:33]: Can we replace 56.0 with `kToolbarHeight` or is it not entirely the same?

File: test/widgets/clean_app_bar_test.dart [hunk 7]
```diff
@@ -0,0 +1,170 @@
+import 'package:collaction_app/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart';
```
Comment (Xazin) [inline test/widgets/clean_app_bar_test.dart:1]: Can we format test files a bit better? Just add `,` wherever it's possible almost, feel the indentation is completely off for both test files! :D

File: test/widgets/clean_app_bar_test.dart [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+          appBar: CleanAppBar(),
+        ),
+      ));
+      expect(
+          find.byWidgetPredicate((widget) =>
+              widget is AppBar && widget.backgroundColor == kSecondaryColor),
+          findsOneWidget);
+    });
+  });
+  group('Tests for centerTitle parameter', () {
```
Comment (Xazin) [inline test/widgets/clean_app_bar_test.dart:65]: Add newline between groups

File: lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart (hunk 5 above)
Comment (Xazin) [reply lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart:54]: https://api.flutter.dev/flutter/material/kToolbarHeight-constant.html

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart (hunk 1 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:59]: Is nested `if`, so I guess using curly braces makes it more readable. What you say?

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart (hunk 3 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:93]: Okay 👌👌

File: lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart (hunk 4 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart:50]: Okay 👌

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart (hunk 1 above)
Comment (Xazin) [reply lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:59]: The comment was for inside the setState function. You cannot use arrow function for returning an if block.

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+    widget.pageScrollController.addListener(() {
+      if (widget.pageScrollController.position.pixels <=
+          scrollController.position.maxScrollExtent) {
+        scrollController.jumpTo(widget.pageScrollController.position.pixels);
+      }
+
+      if (widget.pageScrollController.position.pixels >
+          widget.pageScrollController.position.minScrollExtent) {
+        if (widget.elevation == null) {
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:57]: `if (widget.elevation == null && currentElevation == widget.minElevation) {`
Let's remove the nested if-statement, do the above instead and then only have the `setState(..)` inside of it.

Note: @chinkysight All comments on code in Git, always show a few lines of code, but the specific spot where the comment is made, is always at the last line of code shown.

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+          if (currentElevation == widget.minElevation) {
+            setState(() {
+              currentElevation = widget.maxElevation;
+            });
+          }
+        }
+      } else if (widget.pageScrollController.position.pixels ==
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:64]: Same here, remove the innermost if statement, change this line to be something like:
`} else if (widget.elevation == null && widget.pageScrollController.position.pixels == widget.pageScrollController.position.minScrollExtent) {`
and then just have the `setState(...)` call inside it.

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+              child: Text(widget.title),
+            ),
+          ),
+        ),
+        titleTextStyle: widget.titleTextStyle ??
+            Theme.of(context)
+                .textTheme
+                .headline6
+                ?.copyWith(color: kPrimaryColor),
+        backwardsCompatibility: false,
```
Comment (Xazin) [inline lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:116]: Can we move `backwardsCompatibility` above `titleTextStyle` for cognitive reasons.

**Comment by Xazin** [comment]
Let me know on slack if you have any problems or if you have questions regarding my comments. Left a note on one of the comments trying to explain how git comments work, so you will have an easier time knowing what I'm commenting on in the future. @chinkysight

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart (hunk 1 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:59]: Okay, got it :)

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart (hunk 2 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:67]: Got it 😉

File: lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart (hunk 5 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/clean_app_bar.dart:54]: We can do that

File: lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart (hunk 11 above)
Comment (chinkysight) [reply lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart:116]: Sure thing :)

**Review by Xazin — APPROVED** [review]

**Pull request merged**